src/components/form/components/form_components/input.tsx

```diff
@@ -0,0 +1,68 @@
+import { FormGroup, InputGroup } from '@blueprintjs/core';
+import type { ChangeEvent } from 'react';
+
+import { useInputId } from '../hooks/use_input_id.js';
+
+interface InputProps {
+  error?: string;
+  required?: boolean;
+  id?: string;
+
+  inputGroupProps: {
```

Comment:
I would only put the formGroupProps in a separate property, not the actual input's props.